### PR TITLE
Repository client unit test

### PR DESCRIPTION
**Pull request by bbpennel**
Adds a unit test for the Repository class
Cleans up the API for the class a bit for consistency.

### CODE REVIEW

File: fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectFactory.java [hunk 1]
```diff
@@ -307,8 +307,27 @@ public URI createMemberLink(URI parentUri, URI memberUri) throws FedoraException
...
+
+		try (FcrepoResponse response = getClient().put(uri)
+				.body(RDFModelUtil.streamModel(model), TURTLE_MIMETYPE)
+				.perform()) {
+
+			return response.getLocation();
+		} catch (IOException e) {
+			throw new FedoraException("Unable to create object at " + uri, e);
+		} catch (FcrepoOperationFailedException e) {
+			throw ClientFaultResolver.resolve(e);
```
Comment (harringj) [inline fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectFactory.java]: Is this exception thrown when response.getLocation() fails? Just trying to understand which conditions trigger each of these two exceptions.

File: fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/RepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+		repository.setClient(fcrepoClient);
+
+		PIDs.setRepository(repository);
+	}
+
+	@Test
+	public void mintDepositRecordPidTest() {
+		PID pid = repository.mintDepositRecordPid();
+
+		assertTrue(pid.getRepositoryPath().contains(RepositoryPathConstants.DEPOSIT_RECORD_BASE));
```
Comment (harringj) [inline fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/RepositoryTest.java]: What do you think about asserting something more specific here, say, .startsWith(RepositoryPathConstants.DEPOSIT_RECORD_BASE) ?

File: fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectFactory.java (hunk 1 above)
Comment (bbpennel) [inline fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectFactory.java]: I believe the IOException comes from closing the response, and the FcrepoOperationFailedException is thrown by any call to the fcrepoClient.put/get/post/etc methods, and is thrown whenever a 400 or 500 status code is returned I believe.

File: fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/RepositoryTest.java (hunk 2 above)
Comment (bbpennel) [inline fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/RepositoryTest.java]: Updated this to be a little bit more exact about where the DEPOSIT_RECORD_BASE is in the returned pid

**Pull request merged**